src/js/node/zlib.ts

```diff
@@ -448,19 +748,51 @@ const zlib = {
   BrotliCompress,
   BrotliDecompress,
 
-  ...functions,
+  deflate: createConvenienceMethod(Deflate, false),
+  deflateSync: createConvenienceMethod(Deflate, true),
+  gzip: createConvenienceMethod(Gzip, false),
+  gzipSync: createConvenienceMethod(Gzip, true),
+  deflateRaw: createConvenienceMethod(DeflateRaw, false),
+  deflateRawSync: createConvenienceMethod(DeflateRaw, true),
+  unzip: createConvenienceMethod(Unzip, false),
+  unzipSync: createConvenienceMethod(Unzip, true),
+  inflate: createConvenienceMethod(Inflate, false),
+  inflateSync: createConvenienceMethod(Inflate, true),
+  gunzip: createConvenienceMethod(Gunzip, false),
+  gunzipSync: createConvenienceMethod(Gunzip, true),
+  inflateRaw: createConvenienceMethod(InflateRaw, false),
+  inflateRawSync: createConvenienceMethod(InflateRaw, true),
+  brotliCompress: createConvenienceMethod(BrotliCompress, false),
+  brotliCompressSync: createConvenienceMethod(BrotliCompress, true),
+  brotliDecompress: createConvenienceMethod(BrotliDecompress, false),
+  brotliDecompressSync: createConvenienceMethod(BrotliDecompress, true),
 };
-Object.defineProperty(zlib, "constants", {
-  writable: false,
-  configurable: false,
-  enumerable: true,
-  value: Object.freeze(constants),
-});
-Object.defineProperty(zlib, "codes", {
-  writable: false,
-  configurable: false,
-  enumerable: true,
-  value: Object.freeze(codes),
+
+ObjectDefineProperties(zlib, {
+  createDeflate: createProperty(Deflate),
+  createInflate: createProperty(Inflate),
+  createDeflateRaw: createProperty(DeflateRaw),
+  createInflateRaw: createProperty(InflateRaw),
+  createGzip: createProperty(Gzip),
+  createGunzip: createProperty(Gunzip),
+  createUnzip: createProperty(Unzip),
+  createBrotliCompress: createProperty(BrotliCompress),
+  createBrotliDecompress: createProperty(BrotliDecompress),
+  constants: {
+    enumerable: true,
+    value: ObjectFreeze(constants),
+  },
+  codes: {
+    enumerable: true,
+    value: ObjectFreeze(codes),
+  },
 });
 
+// These should be considered deprecated
+// expose all the zlib constants
+for (const bkey of ObjectKeys(constants)) {
+  if (StringPrototypeStartsWith.$call(bkey, "BROTLI")) continue;
```

Comment:
we could copy paste it just once instead of dynamically checking this every time someone runs `require("zlib")` for the first time?